docs/lang/articles/basic/type.md

```diff
@@ -153,73 +159,88 @@ You may use `ti.cast` to explicitly cast scalar values between different
 
 ```python {2,4}
-u = ti.Vector([2.3, 4.7])
-v = int(u)              # ti.Vector([2, 4])
-# If you are using ti.i32 as default_ip, this is equivalent to:
-v = ti.cast(u, ti.i32)  # ti.Vector([2, 4])
+@ti.kernel
+def foo():
+    u = ti.Vector([2.3, 4.7])
+    v = int(u)              # ti.Vector([2, 4])
+    # If you are using ti.i32 as default_ip, this is equivalent to:
+    v = ti.cast(u, ti.i32)  # ti.Vector([2, 4])
 ```
 
-### Bit casting
+### Bit-casts
 
 Use `ti.bit_cast` to bit-cast a value into another data type. The
 underlying bits will be preserved in this cast. The new type must have
 the same width as the the old type. For example, bit-casting `i32` to
 `f64` is not allowed. Use this operation with caution.
 
+```python
+@ti.kernel
+def foo():
+    a = 3.14
+    b = ti.bit_cast(a, ti.i32) # 1078523331
+    c = ti.bit_cast(b, ti.f32) # 3.14
+```
+
 :::note
 For people from C++, `ti.bit_cast` is equivalent to `reinterpret_cast`.
 :::
 
 ## Compound types
 
-User-defined compound types are created using the `ti.types` module. Supported compound types include vectors, matrices, and structs:
+User-defined compound types can be created using the `ti.types` module. Supported compound types include vectors, matrices, and structs:
 
 ```python
-vec2i = ti.types.vector(2, ti.i32)
-vec3f = ti.types.vector(3, float)
-mat2f = ti.types.matrix(2, 2, float)
-ray = ti.types.struct(ro=vec3f, rd=vec3f, l=ti.f32)
+my_vec2i = ti.types.vector(2, ti.i32)
+my_vec3f = ti.types.vector(3, float)
+my_mat2f = ti.types.matrix(2, 2, float)
+my_ray3f = ti.types.struct(ro=my_vec3f, rd=my_vec3f, l=ti.f32)
 ```
 
 ### Creating fields
 
-Fields of a given compound type can be created with the `.field()` method of a Compound Type:
+Fields of a user-defined compound type can be created with the `.field()` method of a Compound Type:
 
 ```python
-# ti.Vector.field(2, dtype=ti.i32, shape=(233, 666))
-x = vec2i.field(shape=(233, 666))
-
-# ti.Matrix.field(2, 2, dtype=ti.i32, shape=(233, 666))
-x = mat2f.field(shape=(233, 666))
+vec1 = my_vec2i.field(shape=(128, 128, 128))
+mat2 = my_mat2f.field(shape=(24, 32))
+ray3 = my_ray3f.field(shape=(1024, 768))
 
-# ti.Struct.field({'ro': vec3f, 'rd': vec3f, 'l': ti.f32}, shape=(233, 666))
-x = ray.field(shape=(233, 666))
+# is equivalent to:
+vec1 = ti.Vector.field(2, dtype=ti.i32, shape=(128, 128, 128))
+mat2 = ti.Matrix.field(2, 2, dtype=ti.i32, shape=(24, 32))
+ray3 = ti.Struct.field({'ro': my_vec3f, 'rd': my_vec3f, 'l': ti.f32}, shape=(1024, 768))
 ```
 
 ### Creating local variables
-Compound types can be directly called to create matrix or struct instances. Vectors and matrices can be created using GLSL-like broadcast syntax since the shape of the vector or matrix is already known:
+Compound types can be directly called to create vectorm, matrix or struct instances. Vectors, matrices and structs can be created using GLSL-like broadcast syntax since the their shapes are already known:
```

Comment:
```suggestion
Compound types can be directly called to create vector, matrix or struct instances. Vectors, matrices and structs can be created using GLSL-like broadcast syntax since their shapes are already known:
```